### PR TITLE
Fix forms generator bugs

### PR DESCRIPTION
**Pull request by britneywwc**
Note: form template usage [document](https://webteam.canonical.com/practices/automated-form-builder) should be updated once this is merged

## Done

- Add exception checks when rendering form template
- Use form template on `/data/postgresql` modal
- Added arialabel and placeholder for textarea
- Added exception to remove contact us formfield with `id=about-you` to be excluded from `Comments_from_lead__c`
- Remove unused `modal.js` file
- Apply form modal design as approved by Visual team (similar to https://ubuntu.com/aws#get-in-touch)
- Add inputType fields to `forms-data.json` to accept params: `radio`, `checkbox`, `checkbox-visibility`

## QA

- Go to https://canonical-com-1443.demos.haus/data/postgresql
- Check that secondary navigation is showing (reported bug #1399)
- Click on "Get in touch"
- Modal should pop up
- Check that the formfields are similar to prod https://canonical.com/data/postgresql#get-in-touch
- Check that "Tell us about your use case" textarea has aria label and placeholder text
- Fill up form, submit and see that `Comments_from_lead__c` only includes the comments from "Tell us about your use case" text area

## Issue / Card

Fixes [WD-17343](https://warthogs.atlassian.net/browse/WD-17343) and #1399 

## Screenshots

[if relevant, include a screenshot]

[WD-17343]: https://warthogs.atlassian.net/browse/WD-17343?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: static/js/modals.js [hunk 1]
```diff
@@ -190,7 +190,7 @@ function getCheckboxItemsAsCSV(fieldset) {
 function getCustomFields(event) {
   var message = "";
 
-  document.querySelectorAll('fieldset').forEach(function (formField) {
+  document.querySelectorAll('fieldset:not(#about-you)').forEach(function (formField) {
```
Comment (petesfrench) [inline static/js/modals.js:193]: This feels oddly specific for such a general form generation script. Could we not find a more generic way to do this either in the JS or the form itself

File: static/js/modals.js [hunk 2]
```diff
@@ -264,12 +264,14 @@ function getCustomFields(event) {
   });
 
   const howManyMachinesFieldset = document.getElementById("how-many-machines");
-  const machinesInputs = howManyMachinesFieldset?.querySelectorAll(
-    "input[name='how-many-machines-do-you-have']"
-  );
-  machinesInputs.forEach((input) => {
-    input.removeAttribute("name");
-  });
+  if (howManyMachinesFieldset?.length > 0) {
```
Comment (petesfrench) [inline static/js/modals.js:267]: Didn't we remove this field before and make it more generic?

**Comment by petesfrench** [comment]
The design of the form has changed, is this intentional?
Also there seem to be a lack of white space above the form heading
Current:
image
Old:
image

File: static/files/forms-data.json [hunk 3]
```diff
@@ -3,80 +3,24 @@
...
-              "label": "50 - 100 machines"
-            },
-            {
-              "type": "radio",
-              "id": "greater-than-100",
-              "value": "greater than 100",
-              "label": "> 100 machines"
+              "id": "comments",
+              "ariaLabel": "Anything you'd like to communicate about your needs or interests?",
+              "placeholder": "Anything you'd like to communicate about your needs or interests?"
```
Comment (petesfrench) [inline static/files/forms-data.json:23]: A screen reader will read the placeholder, so as they have the same content you can omit the `aria-label`

**Comment by lyubomir-popov** [comment]
Hi, the "old" version is what was designed by visual. The "current version" looks like something that has not undergone visual. Can we simply implement this as agreed? From my records, [this is the final template](https://www.figma.com/design/mxftgCWDJ6s3LrlaedREkP/24.10-Static-forms?node-id=106-1586&t=3T3Tv8uUPHjDd387-1), 
 with a "ready for dev" label on it.

File: static/js/modals.js (hunk 2 above)
Comment (britneywwc) [reply static/js/modals.js:267]: Yeah good catch, we introduced a new `js-remove-radio-names` on u.com. I'll apply this change to the forms :)

**Review by petesfrench — APPROVED** [review]
LGTM! 

Just one q, why is there now a 'how many machines?' question in the form?

**Comment by britneywwc** [comment]
Thanks for the +1 @petesfrench! I will remove that field before merging, it was added just for QA purposes :)

**Pull request merged**